### PR TITLE
PCRM - 3634 -  System Tests 

### PR DESCRIPTION
**Pull request by premnhs**

### CODE REVIEW

File: .github/workflows/system-test.yml [hunk 1]
```diff
@@ -34,8 +34,29 @@ jobs:
...
-
+        cat output.json
+        echo "Running System tests"
+        pip install -r test_requirements.txt
+        pip install -e .
+        curl -SsL https://downloads.gauge.org/stable | sh
+        cd test/integrationtests
+        gauge run specs
+    - name: upload Test Results artifact
+      if: always()
```
Comment (josh-scargill-nhsd) [inline .github/workflows/system-test.yml:50]: Is this right? looks like a `while True` almost :D

File: terraform/environment/dev/main.tf [hunk 2]
```diff
@@ -8,7 +8,7 @@ terraform {
 }
 
 provider "aws" {
-  region  = "eu-west-2"
+  region = "eu-west-2"
```
Comment (josh-scargill-nhsd) [inline terraform/environment/dev/main.tf:11]: this isn't aligned now

File: terraform/environment/dev/output.tf [hunk 3]
```diff
@@ -22,6 +22,77 @@ output "patients_queue" {
...
+  value = module.List-Recon.LR_08_Lambda
+}
+
+output "lr_08_lambda_arn" {
+  value = module.List-Recon.LR_08_Lambda_arn
+}
+
+output "lr_10_sfn" {
+  value = module.List-Recon.LR_10_SFN
+}
```
Comment (josh-scargill-nhsd) [inline terraform/environment/dev/output.tf:51]: Do you need LR-09 too?

File: terraform/environment/dev/variable.tf [hunk 4]
```diff
@@ -8,6 +8,9 @@ variable "runtime" {
   default = "python3.8"
 }
 
+variable "region" {
+  default = "eu-west-2"
+}
 variable "lambda_timeout" {
```
Comment (josh-scargill-nhsd) [inline terraform/environment/dev/variable.tf:14]: new line

File: test/integrationtests/step_impl/tf_aws_resources.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+import os
+
+cwd = os.path.dirname(__file__)
+OUTPUT_FILE = os.path.join(cwd, "..", "..", "..", "output.json")
+
+
+def get_aws_resources():
+    with open(OUTPUT_FILE) as json_data:
+        tf_output = json.load(json_data)
+        return tf_output
```
Comment (josh-scargill-nhsd) [inline test/integrationtests/step_impl/tf_aws_resources.py:11]: Should this file be a util instead?

File: .github/workflows/system-test.yml (hunk 1 above)
Comment (aparimoo-nhs) [reply .github/workflows/system-test.yml:50]: Yes, This ensures the logs and report get uploaded even if tests fail

File: test/integrationtests/step_impl/tf_aws_resources.py (hunk 5 above)
Comment (aparimoo-nhs) [reply test/integrationtests/step_impl/tf_aws_resources.py:11]: It is only used in test, i would prefer to keep it next to tests.

**Review by aparimoo-nhs — APPROVED** [review]

File: terraform/environment/test/output.tf [hunk 6]
```diff
@@ -21,3 +21,71 @@ output "lr_01_bucket_retry" {
...
+}
+
+output "lr_08_lambda" {
+  value = module.List-Recon.LR_08_Lambda
+}
+
+output "lr_08_lambda_arn" {
+  value = module.List-Recon.LR_08_Lambda_arn
+}
+
```
Comment (HarveyNHS) [inline terraform/environment/test/output.tf:48]: Do you need LR-09 in this?

File: terraform/modules/list-rec/output.tf [hunk 7]
```diff
@@ -24,6 +24,78 @@ output "LR_01_Bucket_inbound" {
...
+output "Demographics_Difference_Table" {
+  value = module.Demographics_Differences_Table.dynamo_table_name
+}
+
+output "Jobs_Status_Table" {
+  value = module.Jobs_Stats_Table.dynamo_table_name
+}
+
+output "Statuses_Table" {
+  value = module.Statuses_Table.dynamo_table_name
```
Comment (josh-scargill-nhsd) [inline terraform/modules/list-rec/output.tf:99]: 🚌

**Review by josh-scargill-nhsd — APPROVED** [review]

**Review by HarveyNHS — COMMENTED** [review]
Added one comment

**Pull request merged**